--- a/backend/drm/drm.c
+++ b/backend/drm/drm.c
@@ -186,11 +186,8 @@
 	for (size_t i = 0; i < drm_plane->count_formats; ++i) {
-		// Force a LINEAR layout for the cursor if the driver doesn't support
-		// modifiers
 		wlr_drm_format_set_add(&p->formats, drm_plane->formats[i],
 			DRM_FORMAT_MOD_LINEAR);
-		if (type != DRM_PLANE_TYPE_CURSOR) {
-			wlr_drm_format_set_add(&p->formats, drm_plane->formats[i],
-				DRM_FORMAT_MOD_INVALID);
-		}
+		wlr_drm_format_set_add(&p->formats, drm_plane->formats[i],
+			DRM_FORMAT_MOD_INVALID);
 	}
